integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/app/ApiExceptionHandler.java

```diff
@@ -2,45 +2,87 @@
 
 import java.time.Clock;
 import javax.servlet.http.HttpServletRequest;
+import javax.validation.ConstraintViolationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 import uk.gov.crowncommercial.esourcing.integration.server.api.ApiUtil;
 
 /*
  * Catches any errors raised by the API layer and returns a suitable error response.
  */
 @ControllerAdvice(basePackageClasses = ApiUtil.class)
-public class ApiExceptionHandler extends ResponseEntityExceptionHandler {
+public class ApiExceptionHandler extends ResponseEntityExceptionHandler
+ {
 
   private static final Logger LOG = LoggerFactory.getLogger(ApiExceptionHandler.class);
 
-  @Autowired
-  private Clock clock;
+  @Autowired private Clock clock;
 
   /**
    * Catch/mop up everything else and return a generic Internal Server Error status code.
-   * 
+   *
    * @param t a throwable not handled elsewhere
    * @param request servlet request used to retrieve request info such as path
    * @return error response entity
    */
   @ExceptionHandler({Throwable.class})
   public ResponseEntity<Object> handleThrowable(Throwable t, HttpServletRequest request) {
-    
-    LOG.warn("Unhandled exception when handling REST call to {}", request.getPathInfo(), t);
-    
-    HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
-    String path = request.getPathInfo();
-    
-    return new ResponseEntity<Object>(
-        ErrorResponse.builder().timestamp(clock.instant()).status(status.value())
-            .error(status.getReasonPhrase()).message("Unhandled exception").path(path).build(),
+
+    if (t instanceof ConstraintViolationException) {
+      LOG.warn("Constraint violation - {}", t.getMessage());
+
+      HttpStatus status = HttpStatus.BAD_REQUEST;
+
+      return new ResponseEntity<>(
+          ErrorResponse.builder()
+              .timestamp(clock.instant())
+              .status(status.value())
+              .error(status.getReasonPhrase())
+              .message(String.format("Constraint violation - %s", t.getMessage()))
```

Comment:
I have had issues from pen tests before with this even though for a REST API it is fine. The auto mated tools complain if the error they inject gets returned to them as it can lead to XSS issues with JavaScript. E.g. if they inject some javascript, the server side validation says can't validate value "invalid javascript" and then you return it in the response then there's the issue. Best to log and return a message without the exception message.

Reply:
So not specify with params are invalid?